supportedBackends/local/src/main/scala/cromwell/backend/impl/local/LocalBackend.scala

```diff
@@ -1,29 +1,227 @@
+  }
+
+  lazy val runsOnDocker = runtimeAttributes.dockerImage.isDefined
+  lazy val processArgs = {
+    val dockerRun = runtimeAttributes.dockerImage map buildDockerRunCommand getOrElse ""
+    Command(Seq("/bin/bash", "-c", s"cat ${jobPaths.script} | $dockerRun /bin/bash <&0"))
+  }
+
+  // Stream Writers
+  lazy val stdoutWriter = jobPaths.stdout.untailed
+  lazy val stderrTailed = jobPaths.stderr.tailed(100)
+
+  override def preStart() = {
+    // workflowPaths.workflowRoot.createDirectories() TODO move to initialize actor
+    jobPaths.callRoot.createDirectories()
+  }
 
   /**
     * Execute a new job.
     */
-  override def execute: Future[BackendJobExecutionResponse] = ???
+  override def execute: Future[BackendJobExecutionResponse] = {
 
-  override protected def jobDescriptor: BackendJobDescriptor = ???
+    val instantiatedCommand = {
+      def toDockerPath(path: WdlValue): WdlValue = path match {
+        case file: WdlFile => WdlFile(jobPaths.toDockerPath(Paths.get(path.valueString)).toAbsolutePath.toString)
+        case v => v
+      }
+      val pathTransformFunction: WdlValue => WdlValue = if (runsOnDocker) toDockerPath else identity
 
-  override protected def configurationDescriptor: BackendConfigurationDescriptor = ???
+      // Inputs coming from the workflow inputs (json input mapping)
+      val workflowInputEntries = workflowDescriptor.inputs collect {
+        case(fqn, value) if splitFqn(fqn)._1 == call.fullyQualifiedName => splitFqn(fqn)._2 -> value
+      }
+
+      // Inputs coming from the "input" keyword in the workflow declaration. These need to be evaluated because they're expressions
+      val evaluatedInputMappings = call.inputMappings mapValues evaluate
+
+      TryUtils.sequenceMap(evaluatedInputMappings, "Job Input evaluation") flatMap { inputs =>
+        val localizedInputs = localizeInputs(jobPaths, runsOnDocker, fileSystems, inputs ++ workflowInputEntries)
+        call.task.instantiateCommand(localizedInputs, callEngineFunction, pathTransformFunction)
+      }
+    }
+
+    instantiatedCommand match {
+      case Success(command) =>
+        logger.info(s"`$command`")
+        writeScript(command, if (runsOnDocker) jobPaths.callDockerRoot else jobPaths.callRoot)
+        logger.info(s"command: $processArgs")
+        process = Option(processArgs.argv.run(ProcessLogger(stdoutWriter writeWithNewline, stderrTailed writeWithNewline)))
+        // This can't run on the context.dispatcher EC because it blocks incoming message processing (in particular abort)
+        // TODO use a new separate, backend scoped EC ?
+        waitAndPostProcess(scala.concurrent.ExecutionContext.global)
+      case Failure(ex) => Future.successful(BackendJobExecutionFailedResponse(jobDescriptor.key, ex))
+    }
+  }
 
   /**
     * Abort a running job.
     */
-  override def abortJob: Future[JobAbortResponse] = ???
+  override def abortJob: Future[JobAbortResponse] = {
+    Future {
```

Comment:
Yeah I'm not 100% sure of what process.destroy() is doing, if it's blocking or just sending SIGTERM and returning. I'll look into it but if it's unnecessary I'll remove the `Future`